src/sweeps/bayes_search.py

```diff
@@ -302,7 +303,14 @@ def next_sample(
         best_test_X_index = np.argmax(prob_of_improve)
     else:
     """
-    best_test_X_index = np.argmax(e_i)
+
+    # Check to see that the metric varies accross the samples
+    if np.all(np.isclose(filtered_y, filtered_y[0], atol=METRIC_VALUE_ATOL)):
```

Comment:
This is tricky, but here we want to check the gaussian process length scale and not that all y values are close to each other. With this we could get into a situation where we have only a few y values that were input by the user initially that are all close to each other, but the intervening parameter space has different values that vary on scales << 1. We should pull the kernel out of the fitted GP and check the length scale in each dimension, and just check that it is not above some threshold